### PR TITLE
BUG: Fix IsInsideObjectSpace function in PolygonSpatialObject

### PR DESCRIPTION
**Pull request by aylward**
The new implementation now defaults to "IsClosed" being true and
appropriately computes IsInside.  Previously variables (node1 and
node2) were not correctly initialized.

### CODE REVIEW

**Comment by aylward** [comment]
Addresses discourse topic:
https://discourse.itk.org/t/itk5-bug-in-itkpolygonspatialobject-spatialobjecttoimage3-cxx/1774

File: Examples/Filtering/SpatialObjectToImage3.cxx [hunk 1]
```diff
@@ -166,7 +166,9 @@ int main( int argc, char *argv[] )
     point = center + radial;
     polygonPoint.SetPositionInObjectSpace( point );
     polygon->GetPoints().push_back( polygonPoint );
+    std::cout << point[0] << ", " << point[1] << std::endl;
```
Comment (fbudin69500) [inline Examples/Filtering/SpatialObjectToImage3.cxx]: Is this debugging left-over?

Comment (fbudin69500) [reply Examples/Filtering/SpatialObjectToImage3.cxx]: Looking below in the code, I see that there are other `std::cout` calls. If this one is here to stay, maybe a description of what is printed should be added (see line 175).

Comment (fbudin69500) [reply Examples/Filtering/SpatialObjectToImage3.cxx]: I will remove this line from the PR and merge the PR. If the `std::cout` was intended, please create a new PR to add it back.

**Comment by fbudin69500** [comment]
Superseeded by #752.

**Pull request closed without merging**